crates/apub/src/protocol/activities/voting/vote.rs

```diff
@@ -49,12 +54,18 @@ impl From<&VoteType> for i16 {
 
 impl InCommunity for Vote {
   async fn community(&self, context: &Data<LemmyContext>) -> LemmyResult<ApubCommunity> {
-    let community = self
-      .object
-      .dereference(context)
-      .await?
-      .community(context)
-      .await?;
-    Ok(community)
+    let community_id = match self.object.dereference(context).await? {
+      Either::Left(p) => p.community_id,
+      Either::Right(c) => {
+        Post::read(&mut context.pool(), c.post_id)
+          .await?
+          .community_id
+      }
+    };
+    Ok(
+      Community::read(&mut context.pool(), community_id)
+        .await?
+        .into(),
+    )
```

Comment:
It'd be better to either read the postview or commentview, then get the `Community` from that view object. Then you won't have to do multiple reads.